samples/step-by-step/sample.md

```diff
@@ -7,100 +7,191 @@ redirects:
 
-The `Subscriber` project needs notification of a successful order. Hence it subscribes to `OrderPlaced` events from `Server`.
+Like the client and the server, the Subscriber application also needs to be configured as an NServiceBus endpoint.
+
+In the Subscriber application, add the following code to the program's Main method:
+
+snippet:StepByStep-SubscriberInit
+
+This is almost identical to the Server configuration, except for the different endpoint name.
+
+Next, create a message handler for the `OrderPlaced` event. Note that whether handling a command or an event, the syntax for a message handler remains the same.
 
+Create a new class in the Subscriber project named `OrderCreatedHandler` using the following code:
 
-### Subscription Configuration
+snippet:StepByStep-OrderCreatedHandler
 
-The subscription is done in the `app.config` of the project
+The handler only logs the fact that the message was received. In a real system, a subscriber to `OrderPlaced` could charge a credit card for the order, start to prepare the order for shipment, or even update information in a customer loyalty database to track when the customer is eligible for different purchase-driven incentives. The fact that these disparate tasks can be accomplished in completely separate message handlers instead of one monolithic process is one of the many strengths of the Publish/Subscribe messaging pattern.
 
-snippet: subscriptionConfig
+Next, the message publisher must be notified that the subscriber is interested in being notified when `OrderPlaced` events are published. To do that, [subscriptions need to be configured in XML](/nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md).
 
+In the Subscriber application, create an App.config file and add the following XML configuration:
 
-### Handling the event
+snippet:StepByStep-subscriptionConfig
 
-When the event is received it will be passed to `OrderCreatedHandler` for processing.
+The important part is in the `MessageEndpointMappings` section. The `add` directive says that for messages of type `OrderPlaced` within the `Shared` assembly, subscription requests should be sent to the `Samples.StepByStep.Server` endpoint, which is the endpoint that is responsible for publishing `OrderPlaced` events.
 
-snippet:OrderCreatedHandler
+When the Subscriber endpoint initializes, it will read this configuration, and knowing that it contains a message handler for `OrderPlaced`, it will send a special subscription message to the `Samples.StepByStep.Server` endpoint. When that endpoint receives the subscription request, it will store it locally (in this sample, using in-memory storage, but in a production system a database would be used) so that when it publishes a message, it can send a copy to every subscriber that expressed interest.
 
 
 ## Running the solution
 
-Run the solution.
+If following along with this tutorial, Visual Studio must be configured so that multiple projects will start when debugging.
```

Comment:
i think it is enough to say "Set Client, Server, and Subscriber to be startup projects"